control-plane/agents/ha-node/src/main.rs

```diff
@@ -40,23 +72,47 @@ impl Cli {
 
 #[tokio::main]
 async fn main() {
-    let cli = Cli::args();
+    let cli_args = Cli::args();
+
+    utils::print_package_info!();
+
+    utils::tracing_telemetry::init_tracing(
+        "agent-ha-node",
+        cli_args.tracing_tags.clone(),
+        cli_args.jaeger.clone(),
+    );
 
     CLUSTER_AGENT_CLIENT
-        .set(ClusterAgentClient::new(cli.cluster_agent, None).await)
+        .set(ClusterAgentClient::new(cli_args.cluster_agent.clone(), None).await)
         .ok()
         .expect("Expect to be initialized only once");
 
-    cluster_agent_client()
+    if let Err(e) = cluster_agent_client()
```

Comment:
registration logic will change for sure (more to 'heartbeat' semantics).
For now it's just to avoid panics upon agent start in case cluster agent is not running.